spec/services/stats/r103_sar_business_unit_performace_report_spec.rb

```diff
@@ -115,38 +123,214 @@ module Stats
+                %q{Performance %,Total received,Responded - in time,Responded - late,Open - in time,Open - late,} +
+                %q{Performance %,Total received,Responded - in time,Responded - late,Open - in time,Open - late,} +
+                %q{Performance %,Total received,Responded - in time,Responded - late,Open - in time,Open - late}
+              expected_text = <<~EOCSV
+                Business unit report (SARs) - 5 Jun 2017 to 30 Jun 2017
+                #{super_header}
+                #{header}
+                BGAB,"","",#{@bizgrp_ab.team_lead},"","",40.0,7,2,0,2,3,0.0,0,0,0,0,0,40.0,7,2,0,2,3,0.0,7,0,0,7,0
+                BGAB,DRA,"",#{@dir_a.team_lead},"","",33.3,4,1,0,1,2,0.0,0,0,0,0,0,33.3,4,1,0,1,2,0.0,4,0,0,4,0
+                BGAB,DRA,RTA,#{@team_a.team_lead},"","",33.3,4,1,0,1,2,0.0,0,0,0,0,0,33.3,4,1,0,1,2,0.0,4,0,0,4,0
+                BGAB,DRB,"",#{@dir_b.team_lead},"","",50.0,3,1,0,1,1,0.0,0,0,0,0,0,50.0,3,1,0,1,1,0.0,3,0,0,3,0
+                BGAB,DRB,RTB,#{@team_b.team_lead},"","",50.0,3,1,0,1,1,0.0,0,0,0,0,0,50.0,3,1,0,1,1,0.0,3,0,0,3,0
+                BGCD,"","",#{@bizgrp_cd.team_lead},"","",100.0,2,1,0,1,0,0.0,0,0,0,0,0,100.0,2,1,0,1,0,0.0,2,0,0,2,0
+                BGCD,DRCD,"",#{@dir_cd.team_lead},"","",100.0,2,1,0,1,0,0.0,0,0,0,0,0,100.0,2,1,0,1,0,0.0,2,0,0,2,0
+                BGCD,DRCD,RTC,#{@team_c.team_lead},"","",100.0,1,1,0,0,0,0.0,0,0,0,0,0,100.0,1,1,0,0,0,0.0,1,0,0,1,0
+                BGCD,DRCD,RTD,#{@team_d.team_lead},"","",0.0,1,0,0,1,0,0.0,0,0,0,0,0,0.0,1,0,0,1,0,0.0,1,0,0,1,0
+                Total,"","","","","",50.0,9,3,0,3,3,0.0,0,0,0,0,0,50.0,9,3,0,3,3,0.0,9,0,0,9,0
+              EOCSV
+              report = R103SarBusinessUnitPerformanceReport.new(period_start: Date.new(2017, 6, 5), period_end: Date.today, generate_bu_columns: true)
+              report.run
+              actual_lines = report.to_csv.map { |row| row.map(&:value) }
+              expected_lines = expected_text.split("\n")
+              actual_lines.zip(expected_lines).each do |actual, expected|
+                expect(CSV.generate_line(actual).chomp).to eq(expected)
+              end
             end
           end
         end
+  
+  
+        context 'with a case in the db that is unassigned' do
+          before do
+            Timecop.freeze Time.new(2017, 6, 30, 12, 0, 0) do
+              create :case, identifier: 'unassigned case'
+            end
+          end
+  
+          it 'does not raise an error' do
+            report = R103SarBusinessUnitPerformanceReport.new
+            expect { report.run }.not_to raise_error
+          end
+        end
+      end
+    end
 
-
-        if type.present?
-          refusal_reason = CaseClosure::RefusalReason.__send__(type)
-          kase.refusal_reason = refusal_reason
+    #rubocop:disable Metrics/ParameterLists
+    def create_case(received:, responded:, deadline:, team:, responder:, ident:, flagged: false, type: nil)
```

Comment:
Not saying that you need to do this but we could use a builder to stop having to pass so many things into this method - just a thought 👍 

Builder pattern: https://medium.com/kkempin/builder-design-pattern-in-ruby-dfa2d557ff1b

Reply:
@willmcb, 👍 ,   had a read the link, it is definitely much clearer way to build the instance  could I do this next time, I will assume we will do more report testing when we do complaint---> sounds a execute 😊 ,  but I ensure I have put this idea in my mind